src/main/ruby/core/process_mirror.rb

```diff
@@ -195,7 +195,7 @@ def initialize(env_or_cmd, *args)
                 from = convert_io_fd key.first
                 to = convert_to_fd value, from, @files_for_child
                 redirect @options, from, to
-                key[1..-1].each { |k| redirect @options, convert_io_fd(k), from }
+                key[1..-1].each { |k| redirect @options, convert_io_fd(k), to }
```

Comment:
I think all `key` can be treated the same now, no need to extract the first one as a special case.
The comment above also needs to be adapted.